mangadex_downloader/upscale_coreml.py

```
@@ -0,0 +1,408 @@
+
+        try:
+            with tqdm(unit='B', unit_scale=True, unit_divisor=1024,
+                      miniters=1, desc="Downloading Core ML model") as pbar:
+                def reporthook(block_num, block_size, total_size):
+                    if pbar.total is None and total_size > 0:
+                        pbar.total = total_size
+                    downloaded = block_num * block_size
+                    if downloaded <= total_size:
+                        pbar.update(block_size)
+
+                urllib.request.urlretrieve(MODEL_URL, zip_path, reporthook=reporthook)
```

Ruff (0.13.3)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)